### PR TITLE
Two JIRAs around Paging and AMQP

### PR DESCRIPTION
**Pull request by clebertsuconic**
These two JIRAs are related but I wanted to keep them separate as commits.

Both changes were part of the same testing.

### CODE REVIEW

File: artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java [hunk 1]
```diff
@@ -99,45 +98,23 @@ public void shutdown(long timeout, TimeUnit unit) {
...
-               //give other threads a chance to make progress
-               LockSupport.parkNanos(100_000_000L);
-            }
-         }
-         while (!stateUpdater.compareAndSet(this, STATE_NOT_RUNNING, STATE_FORCED_SHUTDOWN));
-         //this could happen just one time: the forced shutdown state is the last one and
-         //can be set by just one caller.
-         //As noted on the execute method there is a small chance that some tasks would be enqueued
+      if (!inHandler()) {
+         flush(1, TimeUnit.SECONDS);
```
Comment (franz1981) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java]: I would use some named const somewhere to help understand why is 1 second or I won't wait at all

File: artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java [hunk 2]
```diff
@@ -99,45 +98,23 @@ public void shutdown(long timeout, TimeUnit unit) {
...
-      //there is a small chance that execute() could race with this cleanup: the lock allow an all-or-nothing behaviour between them
-      synchronized (tasks) {
-         T item;
-         while ((item = tasks.poll()) != null) {
-            onPendingItem.accept(item);
-            pendingItems++;
-         }
+
+      T item;
+      while ((item = tasks.poll()) != null) {
```
Comment (franz1981) [inline artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java]: the synchronize was there for 2 reasons: 

1. if we would have started using some multi-producer single-consume queue (the ones on JCTools for example...) 
2. onAddedTaskIfNotRunning is using a synchronize to allow all-or-nothing behaviour

File: artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContextTest.java [hunk 3]
```diff
@@ -154,6 +160,8 @@ private void doOnMessageWithDeliveryException(List<Symbol> sourceSymbols,
 
       rc.onMessage(mockDelivery);
 
+      Thread.sleep(1000);
```
Comment (franz1981) [inline artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContextTest.java]: that's safe in any env?

File: artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/OperationContextImpl.java [hunk 4]
```diff
@@ -43,6 +44,8 @@
  */
 public class OperationContextImpl implements OperationContext {
 
+   private static final Logger logger = Logger.getLogger(OperationContextImpl.class);
```
Comment (brusdev) [inline artemis-server/src/main/java/org/apache/activemq/artemis/core/persistence/impl/journal/OperationContextImpl.java]: is this logger used?

File: artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContextTest.java (hunk 3 above)
Comment (clebertsuconic) [reply artemis-protocols/artemis-amqp-protocol/src/test/java/org/apache/activemq/artemis/protocol/amqp/proton/ProtonServerReceiverContextTest.java]: I forgot to remove this sleep

File: artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java (hunk 1 above)
Comment (clebertsuconic) [reply artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java]: I was afraid of failures on the testsuite. In a production system the component would be going down anyways.


I will try to remove the condition for good.

File: artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java (hunk 2 above)
Comment (clebertsuconic) [reply artemis-commons/src/main/java/org/apache/activemq/artemis/utils/actors/ProcessorBase.java]: I am removing the clear on onAddedTaskifNotRunning

**Comment by franz1981** [comment]
If the few comments are being addressed (when makes sense) I'm ok with this, can you merge it your self?

**Pull request closed without merging**